crypto/src/openssl.rs

```diff
@@ -116,37 +101,70 @@ impl Crypto for OpensslCrypto {
         algs: AlgLen,
         measurement: &Digest,
         info: &[u8],
+        rand_seed: Option<&[u8]>,
     ) -> Result<Self::Cdi, CryptoError> {
         let md = Self::get_digest(algs);
+
+        match rand_seed {
+            Some(seed) => {
+                let args = [
+                    &KdfArgument::KbMode(KdfKbMode::Feedback),
+                    &KdfArgument::Mac(KdfMacType::Hmac(md)),
+                    &KdfArgument::KbSeed(seed),
+                    &KdfArgument::KbInfo(measurement.bytes()),
+                    &KdfArgument::Salt(info),
+                    &KdfArgument::Key(&measurement.bytes()),
+                ];
+                perform_kdf(KdfType::KeyBased, &args, md.size())
+                    .map_err(|_| CryptoError::CryptoLibError)
+            }
+            None => {
+                let args = [
+                    &KdfArgument::KbMode(KdfKbMode::Counter),
+                    &KdfArgument::Mac(KdfMacType::Hmac(md)),
+                    &KdfArgument::KbInfo(measurement.bytes()),
+                    &KdfArgument::Salt(info),
+                    &KdfArgument::Key(&measurement.bytes()),
+                ];
+                perform_kdf(KdfType::KeyBased, &args, md.size())
+                    .map_err(|_| CryptoError::CryptoLibError)
+            }
+        }
```

Comment:
nit: could maybe clean this up by doing something like this. I'm not 100% sure if this works though.

```rust
let mut args = vec![
    &KdfArgument::KbMode(KdfKbMode::Counter),
    &KdfArgument::Mac(KdfMacType::Hmac(md)),
    &KdfArgument::KbInfo(measurement.bytes()),
    &KdfArgument::Salt(info),
    &KdfArgument::Key(&measurement.bytes()),
];

if let Some(seed) = rand_seed {
    args.extend(&KdfArgument::KbSeed(seed));
}

perform_kdf(KdfType::KeyBased, args.as_slice(), md.size())
       .map_err(|_| CryptoError::CryptoLibError)
```

Reply:
Ok, sounds good